src/components/LeafletMarker/index.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Marker, Popup } from 'react-leaflet';
+import L from 'leaflet';
+import PropTypes from 'prop-types';
+import categories from '../../static/categories';
+import './style.scss';
+
+export const LMarker = (props) => {
+  const device = props.device;
+  const deviceIcon = new L.Icon({
```

Comment:
Definition of components should be outside a component's render function. Take this one outside.

Reply:
At second thought, the return value of `L.Icon` isn't a component. Disregard the previous remark.